awx/api/serializers.py

```diff
@@ -4453,6 +4455,294 @@ def validate(self, attrs):
+    char_prompts = serializers.CharField(required=False, write_only=True, allow_blank=False)
+    diff_mode = serializers.CharField(required=False, write_only=True, allow_blank=False)
+    job_tags = serializers.CharField(required=False, write_only=True, allow_blank=False)
+    job_type = serializers.CharField(required=False, write_only=True, allow_blank=False)
+    skip_tags = serializers.CharField(required=False, write_only=True, allow_blank=False)
+    survey_passwords = serializers.CharField(required=False, write_only=True, allow_blank=False)
+    job_slice_count = serializers.IntegerField(required=False, min_value=1)
+    timeout = serializers.IntegerField(required=False, min_value=1)
+
+    class Meta:
+        fields = (
+            'unified_job_template',
+            'identifier',
+            'inventory',
+            'credentials',
+            'limit',
+            'labels',
+            'instance_groups',
+            'execution_environment' 'scm_branch',
+            'verbosity' 'forks' 'char_prompts',
+            'diff_mode',
+            'extra_data',
+            'job_slice_count',
+            'job_tags',
+            'job_type' 'skip_tags',
+            'survey_passwords',
+            'timeout',
+            # these are related objects and we need to add extra validation for them in the parent BulkJobLaunchSerializer
+            #
+        )
+
+
+class BulkJobLaunchSerializer(serializers.Serializer):
+    name = serializers.CharField(max_length=512, write_only=True, required=False)  # limited by max name of jobs
+    jobs = BulkJobNodeSerializer(many=True, allow_empty=False, write_only=True, max_length=1000)
+    description: serializers.CharField(write_only=True, required=False, allow_blank=False)
+    extra_vars: serializers.CharField(write_only=True, required=False, allow_blank=False)
+    organization = serializers.PrimaryKeyRelatedField(
+        queryset=Organization.objects.all(),
+        required=False,
+        default=None,
+        allow_null=True,
+        help_text=_('Inherit permissions from organization roles.'),
+    )
+    # inventory: "",  # Here we can use PrimaryKeyRelatedField so it will automagically do rbac/turn into object
+    limit: serializers.CharField(write_only=True, required=False, allow_blank=False)
+    scm_branch: serializers.CharField(write_only=True, required=False, allow_blank=False)
+    # webhook_service: null,  # Here we can use PrimaryKeyRelatedField so it will automagically do rbac/turn into object, I think, I'm actually not sure how to use this
+    # webhook_credential: null,  # Here we can use PrimaryKeyRelatedField so it will automagically do rbac/turn into object  I think, I'm actually not sure how to use this
+    skip_tags: serializers.CharField(write_only=True, required=False, allow_blank=False)
+    job_tags: serializers.CharField(write_only=True, required=False, allow_blank=False)
+    is_bulk_job: serializers.BooleanField(default=True)
```

Comment:
@jainnikhil30 we shouldn't expose this to user -- we just should set it directly when we instantiate the WorkflowJob